### PR TITLE
Add acmpca to certificatevendor

### PR DESCRIPTION
**Pull request by ts-amz**
# Description
Supersedes the original pr: [https://github.com/aws/aws-connected-device-framework/pull/93](https://github.com/aws/aws-connected-device-framework/pull/93)

Updated to fix the merge from the original PR. Also updated spellings and move utility functions to another file.

## Type of change

- [ ] *Bug fix (non-breaking change which fixes an issue)*
- [x] *New feature (non-breaking change which adds functionality)*
- [ ] *Breaking change (fix or feature that would cause existing functionality to not work as expected)*
- [ ] *Refactor* (existing code being refactored)
- [x] *This change includes a documentation update*

# Submission Checklist

- [ ] CI dry-run passing
- [ ] Integration tests passing locally
- [ ] Change logs generated

### CODE REVIEW

File: source/packages/services/certificatevendor/README.md [hunk 1]
```diff
@@ -4,7 +4,8 @@
 
 The certificate vendor manages the rotation of certificates involving a number of moving parts across CDF and AWS IoT.
 
-There are two flows for certificate rotation. In the fist case, device certificates are pre-created and registered before the rotation request. In this case the device requests a new certificate and is vended an S3 presigned URL in order to download the certificate package. In the second case, the device provides the certificate vendor module a CSR. In this way the device can request an updated certificate while keeping the private key on the device. The certificate vendor then uses a CA certificate registered with AWS IoT to create a device certificate from the CSR and return this certificate to the device.
+There are two flows for certificate rotation. In the fist case, device certificates are pre-created and registered before the rotation request. In this case the device requests a new certificate and is vended an S3 presigned URL in order to download the certificate package. In the second case, the device provides the certificate vendor module a CSR. In this way the device can request an updated certificate while keeping the private key on the device.
```
Comment (canavandl) [inline source/packages/services/certificatevendor/README.md:7]: sp: "fist" -> "first"

File: source/packages/services/certificatevendor/README.md [hunk 2]
```diff
@@ -4,7 +4,8 @@
...
+There are two options for CSR case. One option is to use ACM PCA to issue the certificate, and another option is to create a certificate from a CA certificate registered with AWS IoT, and CA private key sorted in EC2 Parameter store.
```
Comment (canavandl) [inline source/packages/services/certificatevendor/README.md:8]: extra comma: "with AWS IoT, and CA private key" -> "with AWS IoT and CA private key"

File: source/packages/services/certificatevendor/README.md [hunk 3]
```diff
@@ -20,8 +21,41 @@ A certificate package comprising of the certificate, public key and private key
 
 ### Certificate Creation with a device CSR
 
+#### Certiicate Creation with CA private key
```
Comment (canavandl) [inline source/packages/services/certificatevendor/README.md:24]: sp: "Certiicate" -> "Certificate"

**Review by canavandl — APPROVED** [review]
I pointed out a couple of typos but otherwise have no blocking comments

**Review by pcozzi — APPROVED** [review]

**Pull request merged**